tests/conftest.py

```diff
@@ -50,16 +52,34 @@ def file_mediatypes_library():
     return lib
 
 
+@pytest.fixture(scope="session")
+def temp_dir():
+    """Creates a shared library path for tests, and cleans it up after the session."""
+    temp_root = Path(tempfile.gettempdir())
+    test_dir = temp_root / "ts-test-temp"
+
+    if test_dir.exists():
+        shutil.rmtree(test_dir)
+    test_dir.mkdir(parents=True)
```

Comment:
Is there a reason why you wouldn't want to use the `exist_ok` parameter here?
(Tests pass with this on at least Windows and macOS)
```suggestion
    test_dir.mkdir(parents=True, exist_ok=True)
```

Reply:
Just to clear it if it already existed from a failed test prior (I had many goes to get this right 😓) to be thorough, but it can be dropped.